docs/vfio-user.rst

```diff
@@ -488,21 +508,31 @@ format:
 
 Capabilities:
 
-+--------------------+--------+------------------------------------------------+
-| Name               | Type   | Description                                    |
-+====================+========+================================================+
-| max_msg_fds        | number | Maximum number of file descriptors that can be |
-|                    |        | received by the sender in one message.         |
-|                    |        | Optional. If not specified then the receiver   |
-|                    |        | must assume a value of ``1``.                  |
-+--------------------+--------+------------------------------------------------+
-| max_data_xfer_size | number | Maximum ``count`` for data transfer messages;  |
-|                    |        | see `Read and Write Operations`_. Optional,    |
-|                    |        | with a default value of 1048576 bytes.         |
-+--------------------+--------+------------------------------------------------+
-| migration          | object | Migration capability parameters. If missing    |
-|                    |        | then migration is not supported by the sender. |
-+--------------------+--------+------------------------------------------------+
++--------------------+---------+-----------------------------------------------+
+| Name               | Type    | Description                                   |
++====================+=========+===============================================+
+| max_msg_fds        | number  | Maximum number of file descriptors that can   |
+|                    |         | be received by the sender in one message.     |
+|                    |         | Optional. If not specified then the receiver  |
+|                    |         | must assume a value of ``1``.                 |
++--------------------+---------+-----------------------------------------------+
+| max_data_xfer_size | number  | Maximum ``count`` for data transfer messages; |
+|                    |         | see `Read and Write Operations`_. Optional,   |
+|                    |         | with a default value of 1048576 bytes.        |
++--------------------+---------+-----------------------------------------------+
+| migration          | object  | Migration capability parameters. If missing   |
+|                    |         | then migration is not supported by the        |
+|                    |         | sender.                                       |
++--------------------+---------+-----------------------------------------------+
+| cmd_conn           | boolean | Indicates whether the client is capable of    |
```

Comment:
I find this key name a bit unclear. any suggestions for a clearer name?
bidirectional?

Reply:
split_sockets?
half_duplex?
server_socket?
bilateral_sockets?
adjunct_server_socket?
oneway_socket?

I kind of like "half_duplex"... it's not *quite* there but at least has the flavour.

Reply:
aux_socket?

Reply:
I like *split_sockets* since it's closest to capturing the concept IMO). But naming the fd `split_socket_fd` would be confusing then, since it is only one half.

*server_socket* is nice since it indicates what the socket is good for and `server_socket_fd` works well. However, it might be confusing since the socket that the client connects to would generally be referred to as the server socket? And *client_socket* isn't really descriptive, but perhaps *client_cmd_socket*?

Brainstorming a bit more: "twin_socket", "dual_socket", "paired_sockets", "companion_socket", "dyad_socket" (but this is probably too obscure), "client_bound_socket", "reciprocal_socket", "unidirectional_socket". None of these gets me excited though.

Let me know which you like best - English is only a second language to me, so you'll probably be in the better position to make a call.

Reply:
any mention of "client" or "server" is equally ambiguous, client_cmd_socket also sounds like it's commands from the client.

I think "twin_socket" is my favourite for the capability. I'd have something different for the variable name for the actual fd though: "client_sockfd" or something. The latter is obviously a bit less important.

Reply:
I agree `twin_socket` works well as a feature name, so I have adopted that for the capability field and switch to `client_cmd_socket_fd` for the file descriptor.